src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1916,6 +1916,29 @@ void Session::send_message()
     if (m_target_download_mark > m_last_download_mark_sent)
         return send_mark_message(); // Throws
 
+    auto is_upload_allowed = [&]() -> bool {
+        if (!m_is_flx_sync_session) {
+            return true;
+        }
+
+        auto migration_store = get_migration_store();
+        if (!migration_store) {
+            return true;
+        }
+
+        auto sentinel_query_version = migration_store->get_sentinel_subscription_set_version();
```

Comment:
Am I correct that this sentinel will always be set before a `sync::Session` is created and will remain constant throughout the lifetime of the session? Like we create the sentinel in the constructor of `SyncSession` before `m_session` (i.e. this class) is created. So could we just set a member variable of `sync::Session` while it's being constructed and have the sentinel query version throughout the lifetime of the session? That could allow us to avoid having to open a new separate transaction every time we want to send an upload.

Reply:
There is no transaction involved here. `get_sentinel_subscription_set_version` returns a cached value

Reply:
Couldn't the version marked by the sentinel subscription be a moving target if a session is opened but isn't able to connect and then more pending changes are added?

Reply:
I don't think so. Even if the session cannot connect to the server, the user may commit changes as native flx. The subscription set is the limit between migration flx and native flx (and this limit is only set once). There may be a problem if the user switches back to flx (after connecting with native flx), but I don't think we should account for.

Reply:
You're right - any pending changes after natively migrating to FLX "shouldn't" need to worry about the partition value.